glib/ToggleRef.cs

```diff
@@ -69,12 +69,20 @@ public ToggleRef (GLib.Object target)
 			}
 		}
 
+		internal void RemoveSignal (string name)
+		{
+			if (signals == null)
+				return;
+			
+			signals.Remove (name);
+		}
+
   		public void Free ()
   		{
 			if (signals != null) {
-				var array = new Signal [signals.Values.Count];
-				signals.Values.CopyTo (array, 0);
-				foreach (Signal s in array)
+				var copy = signals.Values;
```

Comment:
does invoking `signals.Values` allocate an array whereas iterating over `signals` would not?